tests/scripts/all.sh

```diff
@@ -400,6 +400,18 @@ fi
 
 fi # MemSan
 
+msg "build: cmake build from subdirectory"
+cleanup
+mkdir mbedtls_test_build
```

Comment:
Because this script can be run by anyone, I am not sure it is a good idea to create directories outside the project without the user being aware of it. If you really want to be able to test it outside the mbedtls directory, I suggest we build it in a subdirectory by default and add a command-line option to build it in a user defined directory. Otherwise I would prefer to keep it in this way.